docs/data/data-grid/column-dimensions/column-dimensions.md

```diff
@@ -58,6 +58,29 @@ To capture changes in the width of a column there are two callbacks that are cal
 - `onColumnResize`: Called while a column is being resized.
 - `onColumnWidthChange`: Called after the width of a column is changed, but not during resizing.
 
+## Autosizing [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
+
+`DataGridPro` also allows for autosizing the columns' dimensions based on their content. It can be autosized by 3 methods:
+
+- Double-clicking a column header separator on the grid,
+- Calling the `.autosizeColumns(options)` API method,
+- Adding the `autosizeOnMount` prop.
+
+You can pass options directly to the API method when you call it. If you need to configure autosize for the other two methods, you can provide the options in the `autosizeOptions` prop. Note that for the separator double-click method, the clicked column will be passed to the `.columns` option unless `autosizeOptions` already has a `.columns` list. In all cases, the `.minWidth` and `.maxWidth` options of columns will always be respected.
+
+{{"demo": "ColumnAutosizing.js", "disableAd": true, "bg": "inline"}}
+
+:::warning
+Autosizing has no effect if dynamic row height is enabled.
+:::
+
+:::warning
+The datagrid can only autosize based on the currently rendered cells.
```

Comment:
I think we are using it elsewhere like this
```suggestion
The data grid can only autosize based on the currently rendered cells.
```

Reply:
I prefer it in a single word, feels more reified as a component that way. As opposed to a variant of [grid](https://mui.com/material-ui/react-grid/).

Reply:
Yes, we should standardize it.